### PR TITLE
method_comparison/LoKr: new hyper-params

### PR DESCRIPTION
**Pull request by githubnemo**
After [feedback](https://huggingface.co/papers/2309.14859#6a720049812f2e0c19e69370) of the paper author that the hyper parameter choice for this task was not ideal, I started a search for a better set of hyper parameters and found that given rank 32 a decomposition factor of 4 improves the resuls significantly. No changes to the code base were necessary.

Results targeting `all-linear` instead of q, k, v improved results but wouldn't be comparable. The same goes for the number of steps (LoKr improves results when trained longer, as do other methods).

I also tested the image generation benchmark and here the best result was achieved with `decompose_factor=-1` and tuning the learning rate a bit. `decompose_factor` values of 2 and 4 led to significantly more parameters for comparable (even less) DINO scores.

### CODE REVIEW

**Review by BenjaminBossan — APPROVED** [review]
Thanks for finding these better hyper-params for LoKr.

It's interesting that for image-gen, we don't see any improvements when changing the factor. Still, nice to see a 11 percentage point rise in dino similarity with the same parameter count.

As for MetaMath, we also see a nice increase of 10 percentage points, but that is also with 10x higher learnable parameter count. Beforehand, LoKr was one of the most efficient methods in terms of parameter count, now that's no longer the case. But IMO, memory usage is the more interesting metric anyway, and that value didn't budge.

**Comment by KohakuBlueleaf** [comment]
Yes, usually we mainly tune factor to change the size (the param size is O(M/factor^2) where M is og param size
And best practice is smaller factor (4\~16, depends on model size), and with mid range rank (32~128) or full matrix (no rank)

factor=-1 is usually never optimal but highly depends on task, in anime t2i community we usually use factor8~16 with full rank everywhere. and for large scale finetune a rank4/8 mixed recipe can even reach a "near full scale finetune" result such as Kohaku-XL series model.

The "increase lr" is also a thing, LoKr accept way higher lr than other method, in SDXL finetuning even 1e-2 lr won't make factor=8 explode.
